enterprise/server/testutil/testredis/testredis.go

```diff
@@ -68,12 +74,37 @@ func Start(t *testing.T) string {
 		killed.Store(true)
 		cancel()
 	})
-	return fmt.Sprintf("localhost:%d", redisPort)
+	target := fmt.Sprintf("localhost:%d", redisPort)
+	waitUntilHealthy(t, target)
+	return target
+}
+
+func waitUntilHealthy(t testing.TB, target string) {
+	start := time.Now()
+	ctx := context.Background()
+	r := redis.NewClient(redisutil.TargetToOptions(target))
+	hc := redisutil.HealthChecker{Rdb: r}
+	for {
+		err := hc.Check(ctx)
```

Comment:
nit: i think it's simpler to call `r.Ping(ctx).Err` without needing to instantiate a healthchecker.

Reply:
Done